CRM/Utils/SQL/Select.php

```diff
@@ -323,13 +323,17 @@ public function having($exprs, $args = NULL) {
    *
    * @param string|array $exprs list of SQL expressions
    * @param null|array $args use NULL to disable interpolation; use an array of variables to enable
-   * @return CRM_Utils_SQL_Select
+   * @param int $weight
+   * @return \CRM_Utils_SQL_Select
    */
-  public function orderBy($exprs, $args = NULL) {
+  public function orderBy($exprs, $args = NULL, $weight = NULL) {
     $exprs = (array) $exprs;
+    if ($weight === NULL) {
+      $weight = count($this->orderBys);
```

Comment:
@colemanw This `$weight = count($this->orderBys)` feels weird to me. Proposed fix: https://github.com/colemanw/civicrm-core/pull/4